code/__HELPERS/_lists.dm

```diff
@@ -403,11 +403,28 @@
  * Returns TRUE if the list had nulls, FALSE otherwise
 **/
 /proc/list_clear_nulls(list/list_to_clear)
+	var/static/chain = 1
+	switch(chain)
+		if (1)
+			. = list_clear_nulls_old(arglist(args))
+			if (prob(10))
+				chain = 2
+
+		if (2)
+			. = list_clear_nulls_new(arglist(args))
+			if (prob(10))
+				chain = 1
```

Comment:
I can never get byond to A/B nicely, whenever I test in like a debug world one or the other biases.
Do you have a particular reason for doin it like this?

Reply:
in a real world the proc would take the same codepaths most of the time, especially if ever called in a loop. doing it in a Markov Chain ensures that happens 90% of the time.